File: backend/src/main/java/com/example/backend/repository/FavoriteRepository.java
package com.example.backend.repository;

import com.example.backend.model.FavoriteGame;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Optional;

public interface FavoriteRepository extends JpaRepository<FavoriteGame, Long> {
    Optional<FavoriteGame> findByGameId(Integer gameId);
    void deleteByGameId(Integer gameId);
}

File: backend/src/main/kotlin/com/example/tiendaapp/backend/repository/FavoriteRepository.kt
package com.example.tiendaapp.backend.repository

import com.example.tiendaapp.backend.model.Favorite
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface FavoriteRepository : JpaRepository<Favorite, Long> {
    fun findByGameId(gameId: Int): Favorite?
    fun deleteByGameId(gameId: Int)
}
